src/main/kotlin/org/rust/lang/core/resolve/ref/RsLitExprReferenceContributor.kt

```diff
@@ -62,3 +73,69 @@ private class RsLiteralFileReferenceSet(
         }
     }
 }
+
+private val pathValueLiteral: PsiElementPattern.Capture<RsLitExpr> = psiElement<RsLitExpr>()
+    .with("onPathLiteral") { expr -> expr.kind is RsLiteralKind.String && isPathLitExpr(expr) }
+
+private fun isPathLitExpr(expr: RsLitExpr): Boolean {
+    val (function, arguments) = getFunctionAndArguments(expr) ?: return false
+    val owner = ((function.owner as? RsAbstractableOwner.Impl)?.impl?.typeReference?.type as? TyAdt)?.item
```

Comment:
Let's extract `function.owner as? RsAbstractableOwner.Impl` into local variable to use it in `when` below